spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfiguration.java

```diff
@@ -113,6 +118,20 @@ public KafkaAutoConfiguration(KafkaProperties properties) {
 		return new KafkaTransactionManager<>(producerFactory);
 	}
 
+	@Bean
+	@ConditionalOnProperty(name = "spring.kafka.retry-topic.enabled")
```

Comment:
This does not have any metadata at the moment and it should. See `META-INF/additional-spring-configuration-metadata.properties` for other enabled flags

Reply:
On the other hand why this `enabled` is not a property in that `KafkaProperties.RetryTopic` ?

Reply:
Sure, thanks. I didn't add it there since it's only used in the `@ConditionalOnProperty` annotation and not in the actual configuration code. I'll add it then. I understand that by doing so there's no need for the META-INF properties file entry.